CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+else()
+    # force include Qt's private headers
+    if(NOT "${Qt5_DIR}" STREQUAL "")
+        set(Qt5Core_FIND_COMPONENTS          Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5Gui_FIND_COMPONENTS           Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5OpenGL_FIND_COMPONENTS        Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5Quick_FIND_COMPONENTS         Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5QuickWidgets_FIND_COMPONENTS  Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5Svg_FIND_COMPONENTS           Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5WebEngine_FIND_COMPONENTS     Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5WebEngineCore_FIND_COMPONENTS Private CACHE STRING "force include private headers" FORCE)
+        set(Qt5Widgets_FIND_COMPONENTS       Private CACHE STRING "force include private headers" FORCE)
+    elseif(NOT "${Qt6_DIR}" STREQUAL "")
+        #set(Qt6Core_FIND_COMPONENTS          Private CACHE STRING "force include private headers" FORCE)
+        #set(Qt6Gui_FIND_COMPONENTS           Private CACHE STRING "force include private headers" FORCE)
+        set(Qt6OpenGL_FIND_COMPONENTS        Private CACHE STRING "force include private headers" FORCE)
+        #set(Qt6Quick_FIND_COMPONENTS         Private CACHE STRING "force include private headers" FORCE)
+        #set(Qt6QuickWidgets_FIND_COMPONENTS  Private CACHE STRING "force include private headers" FORCE)
+        #set(Qt6Svg_FIND_COMPONENTS           Private CACHE STRING "force include private headers" FORCE)
+        set(Qt6WebEngine_FIND_COMPONENTS     Private CACHE STRING "force include private headers" FORCE)
+        set(Qt6WebEngineCore_FIND_COMPONENTS Private CACHE STRING "force include private headers" FORCE)
+        set(Qt6WebEngineQuick_FIND_COMPONENTS Private CACHE STRING "force include private headers" FORCE)
+        #set(Qt6Widgets_FIND_COMPONENTS       Private CACHE STRING "force include private headers" FORCE)
+    endif()
+endif()
+
+if("${Qt5_DIR}" STREQUAL "" AND "${Qt6_DIR}" STREQUAL "")
+    message(FATAL_ERROR "Neither Qt5_DIR nor Qt6_DIR defined!")
+else()
+    if(NOT "${Qt5_DIR}" STREQUAL "")
+        find_package(Qt5 REQUIRED COMPONENTS Core Gui OpenGL Quick Quickwidgets Svg Webengine WebengineCore Widgets)
+        set(Qt Qt${Qt5_VERSION_MAJOR}) # TODO find better solution for Qt namespace
+        set(Qt_VERSION       ${Qt5_VERSION})
+        set(Qt_VERSION_MAJOR ${Qt5_VERSION_MAJOR})
+        set(Qt_VERSION_MINOR ${Qt5_VERSION_MINOR})
+        set(Qt_VERSION_PATCH ${Qt5_VERSION_PATCH})
+    elseif(NOT "${Qt6_DIR}" STREQUAL "")
+        find_package(Qt6 REQUIRED COMPONENTS Core Gui OpenGL Quick Quickwidgets Svg WebEngineCore WebEngineQuick Widgets)
+        set(Qt Qt${Qt6_VERSION_MAJOR}) # TODO find better solution for Qt namespace
```

Comment:
@uwerat do you know of a better way to support both `Qt5` and `Qt6` namespace in CMake?

One idea was aliasing imported targets e.g.

```cmake
# alias
add_library(Qt::Core ALIAS Qt5::Core)

# usage
target_link_libraries(${TARGET_NAME} PRIVATE Qt::Core)
```

thereby the CMake syntax stays readable.